mesonbuild/interpreterbase/decorators.py

```diff
@@ -761,6 +762,40 @@ def log_usage_warning(self, tv: str, location: T.Optional['mparser.BaseNode']) -
         mlog.warning(*args, location=location)
 
 
+class FeatureBroken(FeatureCheckBase):
+    """Checks for broken features"""
+
+    # Class variable, shared across all instances
+    #
+    # Format: {subproject: {feature_version: set(feature_names)}}
+    feature_registry = {}  # type: T.ClassVar[T.Dict[str, T.Dict[str, T.Set[T.Tuple[str, T.Optional[mparser.BaseNode]]]]]]
```

Comment:
Why use a type comment?

Reply:
It is copy/pasted from the other subclasses of FeatureCheckBase -- the base uses a type annotation, the subclasses use type comments.

However I realize we do not actually need them... they are inherited...

Reply:
Some day some brave soul will remove all the type comments :)